#! /bin/sh /usr/share/dpatch/dpatch-run
## 14-fix-tcl-flags.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix TCL Tk compilation flags.

@DPATCH@

--- ACE_wrappers/include/makeinclude/platform_linux_common.GNU-	2010-04-30 23:03:53.140594788 +0200
+++ ACE_wrappers/include/makeinclude/platform_linux_common.GNU	2010-04-30 23:09:34.180595413 +0200
@@ -46,24 +46,13 @@
 PLATFORM_BOOST_LDLAGS ?=
 PLATFORM_BOOST_UTF_LIBS ?= -lboost_unit_test_framework
 
-ifeq (Ubuntu, $(findstring Ubuntu,$(LSB_RELEASE_ID)))
-  PLATFORM_TK_CPPFLAGS=$(shell . /usr/lib/tk*/tkConfig.sh && echo -n $$TK_INCLUDE_SPEC $$TK_DEFS)
-  PLATFORM_TK_LIBS=$(shell . /usr/lib/tk*/tkConfig.sh && echo -n $$TK_LIB_FLAG)
-  PLATFORM_TK_LDFLAGS=
-
-  PLATFORM_TCL_CPPFLAGS=
-  PLATFORM_TCL_LIBS=-ltcl8.4
-  PLATFORM_TCL_LDFLAGS=
-else
-  PLATFORM_TK_CPPFLAGS=-I$(shell . /usr/lib*/tkConfig.sh && echo -n $$TK_INC_DIR $$TK_DEFS)
-  PLATFORM_TK_LIBS=$(shell . /usr/lib*/tkConfig.sh && echo -n $$TK_LIB_FLAG)
-  PLATFORM_TK_LDFLAGS=
-
-
-  PLATFORM_TCL_CPPFLAGS=
-  PLATFORM_TCL_LIBS=-ltcl
-  PLATFORM_TCL_LDFLAGS=
-endif
+PLATFORM_TK_CPPFLAGS=$(shell . /usr/lib/tkConfig.sh && echo -n $$TK_INCLUDE_SPEC $$TK_DEFS)
+PLATFORM_TK_LIBS=$(shell . /usr/lib/tkConfig.sh && echo -n $$TK_LIB_FLAG)
+PLATFORM_TK_LDFLAGS=
+
+PLATFORM_TCL_CPPFLAGS=$(shell . /usr/lib/tclConfig.sh && echo -n $$TCL_INCLUDE_SPEC $$TCL_DEFS)
+PLATFORM_TCL_LIBS=$(shell . /usr/lib/tclConfig.sh && echo -n $$(eval echo $$TCL_LIB_FLAG))
+PLATFORM_TCL_LDFLAGS=
 
 PLATFORM_QT_CPPFLAGS ?= -I$(QTDIR)/include
 PLATFORM_QT_LIBS ?= -lqt-mt
